### PR TITLE
txscript: allow script builder capacity to be specified

### PR DESCRIPTION
**Pull request by guggero**
The default script builder allocates 500 bytes of memory for each script. That is quite large for most applications and therefore wastes a lot of memory when only small scripts are created. To avoid breaking backward compatibility, we add a new functional option to the NewScriptBuilder function that allows the user to specify the initial allocation size.

cc @Roasbeef, @Crypt-iQ.

### CODE REVIEW

File: txscript/scriptbuilder_test.go [hunk 1]
```diff
@@ -7,8 +7,38 @@ package txscript
...
 	"bytes"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
+// TestScriptBuilderAlloc tests that the pre-allocation for a script via the
+// NewScriptBuilder function works as expected.
+func TestScriptBuilderAlloc(t *testing.T) {
+	// Using the default value, we should get a script with a capacity of
```
Comment (Roasbeef) [inline txscript/scriptbuilder_test.go:17]: Might be my monitor, but looks to slightly run over line length.

**Review by Roasbeef — APPROVED** [review]
LGTM 🦖

**Review by Crypt-iQ — APPROVED** [review]
nice optimization 🔬

File: txscript/scriptbuilder_test.go (hunk 1 above)
Comment (guggero) [reply txscript/scriptbuilder_test.go:17]: No, it's actually 3 characters short of the full 80 :sweat_smile:

**Pull request merged**